blockly-rtc/src/WorkspaceClient.js

```diff
@@ -108,4 +109,93 @@ export default class WorkspaceClient {
         this.writeInProgress = false;
     };
 
+    /**
+     * Trigger an API call to query events from the database.
+     * @returns {!Array<!Object>} The result of processQueryResults_() or an
+     * empty array if the API call fails.
+     * @public
+     */
+    async queryDatabase() {
+      try {
+        const rows = await getEvents(this.lastSync);
+        return this.processQueryResults_(rows);
+      } catch {
+        return [];
+      };
+    };
+
+    /**
+     * Compare the order of events in the rows retrieved from the database to
+     * the stacks of local-only changes and provide a series of steps that
+     * will allow the server and local workspace to converge.
+     * @param {<!Array<!Object>} rows Rows of event entries retrieved by
+     * querying the database.
+     * @returns {!Array<!Object>} eventQueue An array of events and the
```

Comment:
Can you make these annotations more specific?  Using Object is correct but not very specific.  

You may need to use record types or typedefs.  More information here: https://github.com/google/closure-compiler/wiki/Types-in-the-Closure-Type-System

Reply:
And a nit on annotations: we use `{!Array.<!Object>}`.  Notice the extra period.  Both are valid, so this is just about consistency.